### ISSUE TITLE
Fix shallow cloning of tensors

### ISSUE THREAD

**Issue by Pale-Blue-Dot-97** [Pale-Blue-Dot-97/Minerva#92]
## Issue

`mask` `Tensor` is shallow copied in `modelio.sup_tg`:

```python
y_batch = torch.tensor(masks, dtype=torch.long)
```

## Solution

This must be avoided by using `.clone().detach()` on `masks`.
